src/SapNwRfc/SapServer.cs

```diff
@@ -87,30 +87,31 @@ public void Dispose()
         /// <summary>
         /// Installs a global server function handler.
         /// </summary>
-        /// <param name="connectionString">The connection string.</param>
+        /// <param name="getFunctionMetadata">The metadata handler.</param>
         /// <param name="action">The RFC server function handler.</param>
-        public static void InstallGenericServerFunctionHandler(string connectionString, Action<ISapServerConnection, ISapServerFunction> action)
+        public static void InstallGenericServerFunctionHandler(Func<string, SapAttributes, ISapFunctionMetadata> getFunctionMetadata, Action<ISapServerConnection, ISapServerFunction> action)
         {
-            InstallGenericServerFunctionHandler(SapConnectionParameters.Parse(connectionString), action);
+            InstallGenericServerFunctionHandler(new RfcInterop(), getFunctionMetadata, action);
         }
 
-        /// <summary>
-        /// Installs a global server function handler.
-        /// </summary>
-        /// <param name="parameters">The connection parameters.</param>
-        /// <param name="action">The RFC server function handler.</param>
-        public static void InstallGenericServerFunctionHandler(SapConnectionParameters parameters, Action<ISapServerConnection, ISapServerFunction> action)
-        {
-            InstallGenericServerFunctionHandler(new RfcInterop(), parameters, action);
-        }
+        // We keep a reference to the current server function delegates so they don't get GCed
+        // Otherwise the application crashes as the delegate passed into native does no longer exists in managed
+#pragma warning disable SA1306 // Field names should begin with lower-case letter
+        private static RfcInterop.RfcServerFunction CurrentServerFunction;
```

Comment:
The need is for https://github.com/huysentruitw/SapNwRfc/issues/56 otherwise these delegates get garbage collected and will terminate the application when the function gets called from unmanaged code.